### PR TITLE
Supplememtary easyconfigs for GSL

### PR DESCRIPTION
**Pull request by scintimandrion**
New easyconfigs for GSL with different toolchains.

### CODE REVIEW

**Comment by scintimandrion** [comment]
Yes.

File: easybuild/easyconfigs/g/GSL/GSL-1.16-goolf-1.5.14.eb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+version = '1.16'
+
+homepage = 'http://www.gnu.org/software/gsl/'
+description = """The GNU Scientific Library (GSL) is a numerical library for C and C++ programmers.
+The library provides a wide range of mathematical routines such as random number generators, special functions and least-squares fitting."""
+
+toolchain = {'name': 'goolf', 'version': '1.5.14'}
+toolchainopts = {'opt': True, 'optarch': True, 'unroll': True, 'pic': True}
+
+source_urls = ['http://ftpmirror.gnu.org/gsl/']
```
Comment (wpoely86) [inline easybuild/easyconfigs/g/GSL/GSL-1.16-goolf-1.5.14.eb]: use `GNU_SOURCE` here

**Comment by wpoely86** [comment]
ok to test

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/505bec494c4446138d09 for a full test report.

File: easybuild/easyconfigs/g/GSL/GSL-1.16-goolf-1.5.14.eb (hunk 1 above)
Comment (boegel) [inline easybuild/easyconfigs/g/GSL/GSL-1.16-goolf-1.5.14.eb]: @scintimandrion: please fix this tiny remark, and let us know if you did by issuing a comment

**Comment by scintimandrion** [comment]
The GNU_SOURCE macro has been put in place on all outstanding GSL easyconfigs.

**Comment by wpoely86** [comment]
@boegel ready for merge!

**Comment by boegel** [comment]
Thanks @scintimandrion and @wpoely86 for reviewing! Going in.

**Pull request merged**